src/client/pages/OfferNew/Checkout/index.tsx

```diff
@@ -338,9 +343,15 @@ export const Checkout = ({
   const onPhoneChange = async (phoneNumber: string) => {
     const { phoneNumber: currentPhone } = offerData.person
     if (!phoneNumber || currentPhone === phoneNumber) return
+    setIsPhoneNumberUpdating(true)
     await editQuotes(quoteIds, { phoneNumber })
+    setIsPhoneNumberUpdating(false)
   }
 
+  const isPhoneNumberGoodToGo = isPhoneNumberRequired
+    ? !isPhoneNumberUpdating && typeof offerData.person.phoneNumber === 'string'
```

Comment:
🤔 So, I guess we might have a problem here if we end up in the scenario where `offerData.person.phoneNumber` is an empty string: phone number is required but empty string is acceptable. Maybe we can fix it with:

```
  const isPhoneNumberGoodToGo = isPhoneNumberRequired
    ? !isPhoneNumberUpdating && offerData.person.phoneNumber
    : true
```

Phone number type is `BundledQuote.phoneNumber: string | null | undefined` and `null`, `undefined`, and `''` are falsy values, which would make it pass only in the scenario where `offerData.person.phoneNumber` has some content.

Reply:
hmm, we did that in order to avoid a typing error. if we do not check the type of offerData.person.phoneNumber then it can be undefined and the Sign component does not accept that. 
![image](https://user-images.githubusercontent.com/63097139/150518697-51c2b252-f70a-40b2-bfbd-a2c15b4e40e6.png)

Reply:
I see. In order to `isPhoneNumberGootToGo` to have proper type (`boolean`) we'd need to convert `offerData.person.phoneNumber` to a boolean:

```
  const isPhoneNumberGoodToGo = isPhoneNumberRequired
    ? !isPhoneNumberUpdating && Boolean(offerData.person.phoneNumber)
    : true
```
or even
```
  const isPhoneNumberGoodToGo = isPhoneNumberRequired
    ? !isPhoneNumberUpdating && !!offerData.person.phoneNumber
    : true
```

Yeah, I was thinking on a weird scenario where, for some reason, we get '' from backend initially. Then it would be a problem. But maybe I'm expecting errors where it shouldn't have any.

Reply:
@robinandeer  what do you think ?

Reply:
Hmm yeah I see your point @guilhermespopolin although we need to trust the API for some things. We can't be expected to perform all the checks in the client as well.

I believe it is enough to check if the field is a string. Otherwise, we should solve the root cause in the API (not allow an empty string to be submitted).

Reply:
I agree with you @guilhermespopolin. Isn't what we want to make sure that there is a phone number? I see your point @robinandeer that the root issue is in the API but since we're already performing checks here, as I think we should, it makes sense to me to check for all the possible falsy values (i.e. `undefined`, `null` and `''`). But I might be missing something :)

Personally I like the explicit type conversion `Boolean(offerData.person.phoneNumber)` more than the implicit type coercion `!!offerData.person.phoneNumber` since the latter isn't as immediately obvious IMO.